scipy/signal/dpss.py

```diff
@@ -0,0 +1,181 @@
+from scipy import linalg, fftpack
+
+__all__ = ['dpss']
+
+
+# This code is adapted with permission from nitime (BSD new license)
+
+def dpss(M, NW, Kmax=None, sym=True, norm=None, return_eigvals=False):
+    """
+    Compute the Discrete Prolate Spheroidal Sequences (DPSS).
+
+    This uses the tridiagonal formulation given in [1]_.
+
+    Parameters
+    ----------
+    M : int
+        Window length.
+    NW : float
+        Standardized half bandwidth corresponding to ``2*NW = BW/f0 = BW*N*dt``
+        where ``dt`` is taken as 1.
+    Kmax : int | None, optional
+        Number of DPSS windows to return (orders 0 through Kmax-1).
+        If None (default), return only a single window instead of an
+        array of windows.
+    sym : bool, optional
+        When True (default), generates a symmetric window, for use in filter
+        design.
+        When False, generates a periodic window, for use in spectral analysis.
+    norm : str | None, optional
+        Can be None, "l2", or "inf". If None (default), then
+        "inf" (``max(abs(window))``) is used when ``Kmax is None`` and
+        "l2" (unit norm vectors) otherwise.
+    return_eigvals : bool, optional
+        If True, also return the eigenvalues in addition to the windows.
+
+    Returns
+    -------
+    v : ndarray, shape (Kmax, N) or (N,)
+        The DPSS windows.
+    e : ndarray, shape (Kmax,) or float, optional
+        The eigenvalues for the windows. Only returned if `return_eigvals`
+        evaluates to True.
+
+    References
+    ----------
+    .. [1] Slepian, D. Prolate spheroidal wave functions, Fourier analysis, and
+           uncertainty V: The discrete case. Bell System Technical Journal,
+           Volume 57 (1978), 1371430
+
+    Examples
+    --------
+    We can compare the `kaiser` window to `dpss`, adapted from
+    `here <https://ccrma.stanford.edu/~jos/sasp/Kaiser_DPSS_Windows_Compared.html>`_:
+
+    >>> import matplotlib.pyplot as plt
+    >>> from scipy.signal import dpss, kaiser, freqz
+    >>> N = 51
+    >>> fig, axes = plt.subplots(3, 2, figsize=(5, 7))
+    >>> for ai, alpha in enumerate((1, 3, 5)):
+    ...     win_dpss = dpss(N, alpha)
+    ...     beta = alpha*np.pi
+    ...     win_kaiser = kaiser(N, beta)
+    ...     for win, c in ((win_dpss, 'k'), (win_kaiser, 'r')):
+    ...         win /= win.sum()
+    ...         axes[ai, 0].plot(win, color=c, lw=1.)
+    ...         axes[ai, 0].set(xlim=[0, N-1], title=r'$\\alpha$ = %s' % alpha,
+    ...                         ylabel='Amplitude')
+    ...         w, h = freqz(win)
+    ...         axes[ai, 1].plot(w, 20 * np.log10(np.abs(h)), color=c, lw=1.)
+    ...         axes[ai, 1].set(xlim=[0, np.pi],
+    ...                         title=r'$\\beta$ = %0.2f' % beta,
+    ...                         ylabel='Magnitude (dB)')
+    >>> for ax in axes.ravel():
+    ...     ax.grid(True)
+    >>> axes[2, 1].legend(['DPSS', 'Kaiser'])
+    >>> fig.tight_layout()
+
+    """  # noqa: E501
```

Comment:
Yes it works for PEP8 checkers / linters (for multi-line strings it should come at the end, which makes sense because anywhere else and it would be part of the string)

Reply:
oh, it's for the URL in the docstring?

Reply:
Yep